k9mail-library/src/main/java/com/fsck/k9/mail/internet/MessageExtractor.java

```diff
@@ -76,10 +84,14 @@ public static String getTextFromPart(Part part) {
                      * the stream is now wrapped we'll remove any transfer encoding at this point.
                      */
                     InputStream in = MimeUtility.decodeBody(body);
+                    InputStream possiblyLimitedIn =
+                            textSizeLimit != NO_TEXT_SIZE_LIMIT ? new BoundedInputStream(in, textSizeLimit) : in;
                     try {
-                        return CharsetSupport.readToString(in, charset);
+                        return CharsetSupport.readToString(possiblyLimitedIn, charset);
                     } finally {
                         try {
+                            // This function does magic things for BinaryTempFileBodyInputStream, so the stream passed
+                            // here MUST be the original one obtained from decodeBody above!
                             MimeUtility.closeInputStreamWithoutDeletingTemporaryFiles(in);
```

Comment:
the magic happening there is pretty ugly, but I didn't wanna touch it for this PR :(